### PR TITLE
OCPBUGS-25453: duplicate failure domains in CMPS

### PR DESCRIPTION
**Pull request by vr4manta**
Added check to prevent duplicate failure domains in CPMS.

### CODE REVIEW

**Comment by rvanderp3** [comment]
overall lgtm, can we get a unit test to try to capture this situation?

**Comment by vr4manta** [comment]
> overall lgtm, can we get a unit test to try to capture this situation?

Yep.  Technically the CPMS FD count wasn't checked and would have been caught by the FDs < master count tests, so I enhanced all test to make sure it is correct.  I also added a test for single zone (which really should be no different than FD = 2).

File: pkg/asset/machines/vsphere/machines.go [hunk 1]
```diff
@@ -83,9 +83,11 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 			"machine.openshift.io/cluster-api-machine-type": role,
 		}
 
-		failureDomains = append(failureDomains, machinev1.VSphereFailureDomain{
-			Name: failureDomain.Name,
-		})
+		if !hasFailureDomain(failureDomains, failureDomain.Name) {
```
[the remark this answers is not in the document]
Comment (vr4manta) [reply pkg/asset/machines/vsphere/machines.go:86]: This check is for how we add them to the CPMS.  I can add the use case where user puts same FD name in install-config multiple times into that method.  I'll whip up that validation check.  Thanks.

**Comment by rvanderp3** [comment]
/lgtm
/hold 
placing hold while tests run

**Comment by vr4manta** [comment]
/assign @patrickdillon
Assigning for approval and review

**Pull request merged**